src/coreComponents/mesh/MeshObjectPath.hpp

```diff
@@ -334,8 +334,7 @@ void MeshObjectPath::forObjectsInPath( dataRepository::Group const & meshBodies,
     MeshBody const & meshBody = meshBodies.getGroup< MeshBody >( meshBodyPair.first );
     for( auto const & meshLevelPair : meshBodyPair.second )
     {
-      MeshLevel const & meshLevel = meshBody.getMeshLevel( meshLevelPair.first );
-
+      MeshLevel & meshLevel = meshBody.getMeshLevel( meshLevelPair.first );
```

Comment:
It doesn't compile... The function wasn't used nor tested.

Reply:
The function:
```
template< typename OBJECT_TYPE,
          typename FUNC >
void MeshObjectPath::forObjectsInPath( std::pair< string const, std::map< string, std::vector< string > > > const & levelPair,
                                       MeshLevel const & meshLevel,
                                       FUNC && func ) const
```
didn't have a const ref for MeshLevel... I pushed an another solution with more consts. What to you think ?

Reply:
The new version potentially breaks const-correctness by casting away `const` on objects obtained from a `MeshLevel const &`.

My ideal solution would be adding a separate non-const overload for this function that forwards to the const version:
```c++
template< typename OBJECT_TYPE,
          typename FUNC >
void MeshObjectPath::forObjectsInPath( std::pair< string const, std::map< string, std::vector< string > > > const & levelPair,
                                       MeshLevel & meshLevel,
                                       FUNC && func ) const
{
  forObjectsInPath< OBJECT_TYPE >( levelPair, const_cast< MeshLevel const & >( meshLevel ), [&]( OBJECT_TYPE const & object )
  {
    func( const_cast< OBJECT_TYPE & >(object) );
  } );
}
```
and discard all the `const_cast< OBJECT_TYPE & >` in the const version (but keep those that do `dynamic_cast< OBJECT_TYPE const & >`).

This way the const version only operates on const objects, and only the non-const version casts away constness but does so safely because it starts with a non-const `MeshLevel` in the first place.